### PR TITLE
Mist API tests

### PR DESCRIPTION
**Pull request by evertonfraga**
- `mist.platform`
- `mist.menu.select(id)`

### CODE REVIEW

File: modules/preloader/include/mistAPI.js [hunk 1]
```diff
@@ -159,20 +157,23 @@ module.exports = () => {
                 const filteredId = prefix + filterId(id);
```
Comment (frozeman) [inline modules/preloader/include/mistAPI.js]: Please also add the comments, like the other functions have not above.

File: modules/preloader/include/mistAPI.js [hunk 2]
```diff
@@ -159,20 +157,23 @@ module.exports = () => {
...
                 queue.push({ action: 'selectMenu', id: filteredId });
 
-                for (var e in this.entries) {
+                for (let e in this.entries) {
                     if ({}.hasOwnProperty.call(this.entries, e)) {
                         this.entries[e].selected = (e === filteredId);
```
Comment (frozeman) [inline modules/preloader/include/mistAPI.js]: The other ones need to set `selected = false`, you only seem to set the new one to true, which means multiple ones have true now.

File: modules/preloader/include/mistAPI.js [hunk 3]
```diff
@@ -159,20 +157,23 @@ module.exports = () => {
...
                     }
                 }
+                executeCallback(null, filteredId);
```
Comment (frozeman) [inline modules/preloader/include/mistAPI.js]: Not sure, if calling `select(id)`, should also call the callback for that menu entry..
this api is mainly for updating the menu in mist, not for calling actions in the developers dapp. For this he will have to program his own logic.

The idea would be also that the mist api is not available sometimes, so you wouldn't depend core logic of your app on it. You would want to do it like this:

```js
'click .mySuperButton': function(){

    MyApp.goToPath('/super');

    // also select entry, if mist is available
    if(window.mist) {
        mist.select('superButton');
    }
}
```

**Comment by frozeman** [comment]
Also where is the logic on the backend for selecting the menu entry? I couldn't find another PR and its not in devleop

File: modules/preloader/include/mistAPI.js (hunk 2 above)
Comment (evertonfraga) [inline modules/preloader/include/mistAPI.js]: @frozeman actually this traverses through all entries, where `e === filteredId` returns either `true` or `false`, depending on the current element.

**Comment by evertonfraga** [comment]
@frozeman this logic is here: https://github.com/ethereum/mist/blob/develop/interface/client/mistAPIBackend.js#L126-L130

File: modules/preloader/include/mistAPI.js (hunk 3 above)
Comment (evertonfraga) [inline modules/preloader/include/mistAPI.js]: yeah, that makes sense indeed. updated.

**Comment by evertonfraga** [comment]
I'll first add the 100 menu entries limit before merge. Thanks for the changes

**Pull request merged**